src/c/sknobs.c

```diff
@@ -921,21 +922,32 @@ int sknobs_load(int argc, char *argv[], char *comment) {
 // return 1 on error
 int sknobs_load_string(char *name, char *s, char *comment) {
   // tokenize
+  int return_value = 0;
   int argc = 0;
-  char *argv[MAX_ARGV_SIZE];
+  size_t current_argv_size;
+  char **argv;
   char *token;
   check_sknobs_init();
   if (debug > 2) printf("sknobs_load_string: %s\n", s);
+  argv = malloc(ARGV_SIZE_CHUNK * sizeof(char*));
+  assert(argv);
+  current_argv_size = ARGV_SIZE_CHUNK;
   argv[argc++] = name;
   token = strtok(s, sknobs_delimiters);
   while (token) {
-    assert(argc < MAX_ARGV_SIZE);
+    if(argc >= current_argv_size) {
+      // Incrementally add chunks.
+      assert(argc < MAX_ARGV_SIZE);
```

Comment:
This is OK given the current values of those #defines but abstractly (e.g. if MAX_ARGV_SIZE is not a multiple of ARGV_SIZE_CHUNK) it allows allocation of more than MAX_ARGV_SIZE and it also allows argc to iterate beyond MAX_ARGV_SIZE.  Not horrible but not perfect.

I was going to say "don't worry about it" but I got intrigued and think this works:
```c
    if(argc >= current_argv_size) {
      int newsize = current_argv_size + ARGV_SIZE_CHUNK;
      assert(current_argv_size < MAX_ARGV_SIZE);
      current_argv_size = (newsize <= MAX_ARGV_SIZE) ? newsize : MAX_ARGV_SIZE;
      argv = realloc(argv, current_argv_size * sizeof(char*));
      assert(argv);
    }
```
Or you can punt.  It's not really a practical problem.

Reply:
What about defining MAX to be N chunks?

Reply:
Yep.